### PR TITLE
New devtools layout

### PR DESCRIPTION
**Pull request by jaril**
Picking up the ball from #1056.

image

Followups:
- [ ] Splitbox resizer between editor and viewer doesn't work when it's close to a console message.
- [ ] Remove splitconsole code.
- [ ] Find out why the `SecondaryToolbox` component keeps re-rendering upon message hover.
- [ ] Refactor `selectedPanel` action to adapt it to the secondary toolbox (Console, Elements, Comments).

### CODE REVIEW

**Comment by jaril** [comment]
image (4)

**Comment by jasonLaster** [comment]
A couple comments:

1. we should disable tooltips when hovering on messages
2. we should remove the console close button (X)
3. we should add a media breakpoint similar to what we do in the debugger for the narrow ui [link](https://github.com/RecordReplay/devtools/blob/master/src/devtools/client/debugger/src/components/App.js#L32)
4. we should add a way to toggle to the PM view
5. the header feels too tall now 💅

File: src/ui/components/DevTools.js [hunk 1]
```diff
@@ -44,18 +46,24 @@ function DevtoolsSplitBox({ toolboxExpanded, updateTimelineDimensions, tooltip }
...
-      initialSize={prefs.toolboxHeight}
-      minSize="20%"
-      maxSize="80%"
-      vert={false}
-      onMove={handleMove}
-      startPanel={viewer}
-      endPanel={toolbox}
-      endPanelControl={false}
-    />
+    <>
```
Comment (jasonLaster) [inline src/ui/components/DevTools.js:49]: why do we need this?

File: src/ui/components/Timeline/index.js [hunk 2]
```diff
@@ -116,23 +114,6 @@ export class Timeline extends Component {
...
-      hideTooltip();
-    }
-
-    if (type == "mouseenter") {
-      const time = message.executionPointTime;
-      const offset = this.getPixelOffset(time);
-      setTimelineToMessage({ message, offset });
-    }
-
-    return null;
```
Comment (jasonLaster) [inline src/ui/components/Timeline/index.js:133]: lets highlight the markers

File: src/ui/components/DevTools.js (hunk 1 above)
Comment (jaril) [reply src/ui/components/DevTools.js:49]: Which one are you referring to?

If it's the splitbox, then it's for resizing the left section (primary toolbar, sidebar, editor) and the right section (viewport, secondary toolbar).

File: src/ui/components/SecondaryToolbox/index.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+// This component re-renders every time you hover on a message in the console. Find
+// out why. -jaril
```
Comment (jaril) [inline src/ui/components/SecondaryToolbox/index.js:134]: Todo: 
```
// This component re-renders every time you hover on a message in the console. Find
// out why. -jaril
```

File: src/devtools/client/webconsole/components/Input/JSTerm.js [hunk 4]
```diff
@@ -115,6 +115,7 @@ class JSTerm extends React.Component {
   }
 
   render() {
+    // return null;
```
Comment (jasonLaster) [inline src/devtools/client/webconsole/components/Input/JSTerm.js:118]: ^^^

File: src/ui/components/DevTools.js (hunk 1 above)
Comment (jasonLaster) [reply src/ui/components/DevTools.js:49]: the outer `<>`

**Review by jasonLaster — APPROVED** [review]

**Pull request merged**